packages/components/src/components/buttonlink/buttonlink.styles.ts

```diff
@@ -0,0 +1,62 @@
+import { css } from 'lit';
+
+import { hostFocusRingStyles } from '../../utils/styles';
+
+const styles = css`
+  :host {
+    border: unset;
+  }
+
+  :host([data-btn-type='pill']),
+  :host([data-btn-type='pill-with-icon']) {
+    padding: unset;
+  }
+
+  :host([data-btn-type='pill'][size='24'])::part(anchor),
+  :host([data-btn-type='pill-with-icon'][size='24'])::part(anchor) {
+    padding: 0px 0.625rem;
+  }
+
+  :host([data-btn-type='pill'][size='32'])::part(anchor),
+  :host([data-btn-type='pill-with-icon'][size='32'])::part(anchor),
+  :host([data-btn-type='pill'][size='28'])::part(anchor),
+  :host([data-btn-type='pill-with-icon'][size='28'])::part(anchor) {
+    padding: 0px 0.75rem;
+  }
+
+  :host([data-btn-type='pill'][size='40'])::part(anchor),
+  :host([data-btn-type='pill-with-icon'][size='40'])::part(anchor) {
+    padding: 0px 1rem;
```

Comment:
nitpick: we need not specify the unit when the value is 0
```suggestion
    padding: 0 0.625rem;
  }

  :host([data-btn-type='pill'][size='32'])::part(anchor),
  :host([data-btn-type='pill-with-icon'][size='32'])::part(anchor),
  :host([data-btn-type='pill'][size='28'])::part(anchor),
  :host([data-btn-type='pill-with-icon'][size='28'])::part(anchor) {
    padding: 0 0.75rem;
  }

  :host([data-btn-type='pill'][size='40'])::part(anchor),
  :host([data-btn-type='pill-with-icon'][size='40'])::part(anchor) {
    padding: 0 1rem;
```